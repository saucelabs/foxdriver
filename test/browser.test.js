import Browser from './../lib/browser'

let browser
let getTabActorsSpy

beforeEach(async () => {
    browser = new Browser('localhost', 9222)
    getTabActorsSpy = jest.spyOn(browser, "getTabActors");
})

test('should call getTabActors if consoleActor is missing from listTabs', async () => {
    browser.request = jest.fn(function (message) {
        if (message === 'listTabs') {
            return Promise.resolve({
                preferenceActor: 'server1.conn6.preferenceActor2',
                selected: 0,
                tabs: [ { actor: 'server1.conn6.tabDescriptor1', traits: [Object] } ],
                from: 'root'
            });
        } else if (message === 'getTarget') {
            return Promise.resolve({
                frame: {
                    consoleActor: 'server1.conn2.child1/consoleActor2',
                }
            });
        } else {
            return Promise.resolve({});
        }
    });

    await browser.listTabs()
    expect(getTabActorsSpy).toHaveBeenCalledTimes(1)
})

test('should not call getTabActors if consoleActor presents', async () => {
    browser.request = jest.fn(function (message) {
        if (message === 'listTabs') {
            return Promise.resolve({
                preferenceActor: 'server1.conn6.preferenceActor2',
                selected: 0,
                tabs: [ {
                    actor: 'server1.conn2.child1/frameTarget1',
                    consoleActor: 'server1.conn2.child1/consoleActor2',
                } ],
                from: 'root'
            });
        } else {
            return Promise.resolve({});
        }
    });

    await browser.listTabs()
    expect(getTabActorsSpy).not.toHaveBeenCalled()
})
