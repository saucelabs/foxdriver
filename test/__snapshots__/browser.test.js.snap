// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call getTabActors if consoleActor is missing from listTabs 1`] = `
Array [
  Tab {
    "_actors": Object {},
    "_events": Object {},
    "_eventsCount": 0,
    "_maxListeners": undefined,
    "_registeredDomains": Object {},
    "client": Client {
      "_activeRequests": Object {},
      "_events": Object {
        "end": [Function],
        "error": [Function],
        "message": Array [
          [Function],
          [Function],
          [Function],
        ],
        "timeout": [Function],
      },
      "_eventsCount": 4,
      "_maxListeners": undefined,
      "_pendingRequests": Array [
        Object {
          "callback": [Function],
          "message": Object {
            "to": "server1.conn6.tabDescriptor1",
            "type": "getTarget",
          },
          "to": "server1.conn6.tabDescriptor1",
        },
      ],
      "host": "localhost",
      "incoming": Object {
        "data": Array [],
        "type": "Buffer",
      },
      "log": Object {
        "chrome": [Function],
        "debug": [Function],
        "error": [Function],
        "firefox": [Function],
        "http": [Function],
        "info": [Function],
        "level": "info",
        "levels": Array [
          "silly",
          "verbose",
          "debug",
          "info",
          "http",
          "warn",
          "error",
          "chrome",
          "firefox",
        ],
        "silly": [Function],
        "verbose": [Function],
        "warn": [Function],
      },
      "port": 9222,
      "supportedDomains": Array [],
      Symbol(kCapture): false,
    },
    "data": Object {
      "actor": "server1.conn6.tabDescriptor1",
      "traits": Array [
        [Function],
      ],
    },
    "isEnabled": false,
    "log": Object {
      "chrome": [Function],
      "debug": [Function],
      "error": [Function],
      "firefox": [Function],
      "http": [Function],
      "info": [Function],
      "level": "info",
      "levels": Array [
        "silly",
        "verbose",
        "debug",
        "info",
        "http",
        "warn",
        "error",
        "chrome",
        "firefox",
      ],
      "silly": [Function],
      "verbose": [Function],
      "warn": [Function],
    },
    "name": "server1.conn6.tabDescriptor1",
    Symbol(kCapture): false,
  },
]
`;

exports[`should not call getTabActors if consoleActor presents 1`] = `
Array [
  Tab {
    "_actors": Object {
      "consoleActor": "server1.conn2.child1/consoleActor2",
    },
    "_events": Object {},
    "_eventsCount": 0,
    "_maxListeners": undefined,
    "_registeredDomains": Object {},
    "client": Client {
      "_activeRequests": Object {},
      "_events": Object {
        "end": [Function],
        "error": [Function],
        "message": Array [
          [Function],
          [Function],
        ],
        "timeout": [Function],
      },
      "_eventsCount": 4,
      "_maxListeners": undefined,
      "_pendingRequests": Array [],
      "host": "localhost",
      "incoming": Object {
        "data": Array [],
        "type": "Buffer",
      },
      "log": Object {
        "chrome": [Function],
        "debug": [Function],
        "error": [Function],
        "firefox": [Function],
        "http": [Function],
        "info": [Function],
        "level": "info",
        "levels": Array [
          "silly",
          "verbose",
          "debug",
          "info",
          "http",
          "warn",
          "error",
          "chrome",
          "firefox",
        ],
        "silly": [Function],
        "verbose": [Function],
        "warn": [Function],
      },
      "port": 9222,
      "supportedDomains": Array [],
      Symbol(kCapture): false,
    },
    "data": Object {
      "actor": "server1.conn2.child1/frameTarget1",
      "consoleActor": "server1.conn2.child1/consoleActor2",
    },
    "isEnabled": false,
    "log": Object {
      "chrome": [Function],
      "debug": [Function],
      "error": [Function],
      "firefox": [Function],
      "http": [Function],
      "info": [Function],
      "level": "info",
      "levels": Array [
        "silly",
        "verbose",
        "debug",
        "info",
        "http",
        "warn",
        "error",
        "chrome",
        "firefox",
      ],
      "silly": [Function],
      "verbose": [Function],
      "warn": [Function],
    },
    "name": "server1.conn2.child1/frameTarget1",
    Symbol(kCapture): false,
  },
]
`;
